Chessington.GameEngine/Pieces/Bishop.cs

```diff
@@ -1,36 +1,40 @@
-﻿using System.Collections.Generic;
+﻿using System;
+using System.Collections.Generic;
 using System.Linq;
 
 namespace Chessington.GameEngine.Pieces
 {
     public class Bishop : Piece
     {
-        public Bishop(Player player)
-            : base(player) { }
-
         public override IEnumerable<Square> GetAvailableMoves(Board board)
         {
+            var initialRow = board.FindPiece(this).Row;
+            var initialCol = board.FindPiece(this).Col;
             var availableMoves = new List<Square>();
-            for (int i = -GameSettings.BoardSize; i < GameSettings.BoardSize; i++)
+
+            for (var i = -GameSettings.BoardSize; i <= GameSettings.BoardSize; i++)
             {
-                if ((board.FindPiece(this).Row + i <= GameSettings.BoardSize) && (board.FindPiece(this).Col + i <= GameSettings.BoardSize))
+                if (CheckPosition(initialRow + i, initialCol + i))
                 {
-                    availableMoves.Add(Square.At(board.FindPiece(this).Row + i, board.FindPiece(this).Col + i));
+                    availableMoves.Add(Square.At(initialRow + i, initialCol + i));
                 }
-                else if ((board.FindPiece(this).Row + i <= GameSettings.BoardSize) && (board.FindPiece(this).Col - i >= 0))
+
+                if (CheckPosition(initialRow + i, initialCol - i))
                 {
-                    availableMoves.Add(Square.At(board.FindPiece(this).Row + i, board.FindPiece(this).Col - i));
+                    availableMoves.Add(Square.At(initialRow + i, initialCol - i));
                 }
             }
+            availableMoves.RemoveAll(x => x == Square.At(initialRow,initialCol));
 
             return availableMoves;
         }
 
+        public Bishop(Player player)
+            : base(player) { }
+
         private bool CheckPosition(int row, int col)
```

Comment:
Since this is boolean it is name can be a question like name. smth like "IsValidPosition"